cmd/node/factory/structs.go

```diff
@@ -577,11 +577,6 @@ func prepareGenesisBlock(args *processComponentsFactoryArgs, genesisBlocks map[u
 		if errNotCritical != nil {
 			log.Error("error storing genesis shardblock", "error", errNotCritical.Error())
 		}
-		hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(genesisBlock.GetShardID())
```

Comment:
Actually this storage exists in each shard but only for self shard. In metachain it is created for every shard.

Reply:
But we do not need the shard genesis block stored inside the nonce-hash unit

Reply:
Ok, in this case you can move line 565 inside first if after line 570

Reply:
done